### ISSUE TITLE
searchBetween, searchBefore, searchAfter

### ISSUE THREAD

**Issue by splitice** [unixs/node-btree#17]
It would be extremely useful if this could implement range search functions.

Currently we are using node-bst (https://github.com/HalleyAssist/node-binary-search-tree-continued/) for our nedb fork. I'd love to explore the possibility of using node-btree. Initial experimentation indicate a decent performance improvement could be had.

**Comment by splitice** [comment]
Thoughts?

**searchBefore:**

Pseudocode:
```
node = gtree_search(tree, key)
while(node){
ret.push(node)
node = g_tree_node_previous(node)
}
```

**searchAfter:**

Pseudocode:
```
node = gtree_search(tree, key)
while(node){
ret.push(node)
node = g_tree_node_next(node)
}
```

**searchBetween:**

Pseudocode:
```
node = gtree_search(tree, keyLower)
nodeUpper = gtree_search(tree, keyUpper)
while(node != nodeUpper){
ret.push(node)
node = g_tree_node_next(node)
}
```

**Comment by unixs** [comment]
A good idea!
I will find time for this

**Comment by unixs** [comment]
GLib supports direct node access since version 2.68. Now it can be implemented

**Comment by splitice** [comment]
Due to the time we have less interest but that doesn't diminish it's usefulness

**Comment by unixs** [comment]
Done.
Fixed in v0.7.0
